### PR TITLE
CU-376p5gx - If I use the WalletConnect deep link when the wallet is …

### PR DESCRIPTION
**Pull request by raulduartep**
…not open it should show the connection modal after I login

**What current issue(s) from Trello/Github does this address?**

**What problem does this PR solve?**

**How did you solve this problem?**

**How did you make sure your solution works?**

**Are there any special changes in the code that we should be aware of?**

**Is there anything else we should know?**

- [ ] Unit tests written?

### CODE REVIEW

**Comment by lllwvlvwlll** [comment]
Task linked: [CU-376p5gx If I use the WalletConnect deep link when the wallet is not open it should show the connection modal after I login](https://app.clickup.com/t/376p5gx)

File: app/components/ConnectDapp/ApproveConnection/ApproveConnection.jsx [hunk 1]
```diff
@@ -109,7 +109,10 @@ const ApproveConnection = ({
     <FullHeightPanel
       headerText="Wallet Connect"
       renderCloseButton={() => (
-        <CloseButton onClick={rejectSession} routeTo={ROUTES.DASHBOARD} />
+        <CloseButton
+          onClick={() => rejectSession()}
+          routeTo={ROUTES.DASHBOARD}
+        />
```
Comment (comountainclimber) [inline app/components/ConnectDapp/ApproveConnection/ApproveConnection.jsx:115]: This change is not actually doing anything, why did you add it?

**Comment by comountainclimber** [comment]
@raulduartep from now on we need to include at least a minimal description of what changed, and why as well as a relevant title in PRs to the neon wallet repo. This is an open source project and pasting a link to a clickup issue is not enough. Thanks 👍

**Comment by comountainclimber** [comment]
@melanke Do not merge this without thorough testing on windows, linux and macOS @raulduartep completely rewrote the way deep linking works with this change

**Comment by comountainclimber** [comment]
I have confirmed that this is at least working as expected on MacOS nice work @raulduartep 💪

File: main.js [hunk 2]
```diff
@@ -294,19 +259,37 @@ ipcMain.handle('dialog', async (event, method, params) => {
...
   const win = BrowserWindow.getFocusedWindow()
   win.setFullScreen(!win.isFullScreen())
 })
+
 ipcMain.handle('close', () => {
   const win = BrowserWindow.getFocusedWindow()
   win.close()
 })
 
+ipcMain.handle('restore', () => {
```
Comment (comountainclimber) [inline main.js:283]: 👍

**Review by comountainclimber — APPROVED** [review]
